jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/ui/activities/DetailActivity.kt

```diff
@@ -1,35 +1,44 @@
 package com.e_conomic.weatherapp.ui.activities
 
 import android.os.Bundle
-import android.support.v7.app.AppCompatActivity
 import android.support.v7.widget.Toolbar
+import android.util.Log
 import android.widget.TextView
 import com.e_conomic.weatherapp.R
 import com.e_conomic.weatherapp.domain.model.Forecast
 import com.e_conomic.weatherapp.extensions.color
 import com.e_conomic.weatherapp.extensions.toDateString
-import com.e_conomic.weatherapp.ui.utils.ToolbarManager
+import com.e_conomic.weatherapp.ui.utils.ToolbarAppCompatActivity
 import com.squareup.picasso.Picasso
 import kotlinx.android.synthetic.main.activity_detail.*
 import org.jetbrains.anko.find
 import java.text.DateFormat
 
-class DetailActivity : AppCompatActivity(), ToolbarManager {
+class DetailActivity : ToolbarAppCompatActivity() {
 
     override val toolbar by lazy {find<Toolbar>(R.id.toolbar)}
 
+    val DETAIL_ACTIVITY_TAG = "DetailActivity"
+
     companion object {
         val FORECAST_EXTRA_KEY = "forecast"
         val FORECAST_CITY_EXTRA_KEY = "forecastCity"
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_detail)
-        initToolbar()
+        super.onCreate(savedInstanceState)
```

Comment:
Hmm maybe we can make this even prettier. Let's assume that we would always call `setContentView()` in `onCreate()`. So what we would actually need here is just the `int` value that represents the respective layout (in this case `R.layout.activity_detail`). If we create an abstract value name let's say `abstract val layoutId: Int` that will be requested as a parameter for `setContentView()`  in the abstract class' `onCreate` method, we can still call `super.onCreate` first. What do you think, does it get too complicated to follow?